03_Tasks/MapWorld/maps.py

```diff
@@ -189,11 +189,15 @@ def assign_types(self, target_type_distr):
                                         len(target_type_distr), replace=False)
         for target_type, repetitions in zip(target_types, target_type_distr):
             for _ in range(repetitions):
-                this_node = unassigned[np.random.choice(range(len(unassigned)))]
-                G.nodes[this_node]['base_type'] = 'indoor'
-                G.nodes[this_node]['type'] = target_type
-                G.nodes[this_node]['target'] = True
-                unassigned.remove(this_node)
+                try:
```

Comment:
Could that not also be fixed by checking whether `len(unassigned) > 0`, and skipping if it isn't? Using `try/except` is always a bit inelegant compared to making it actual program logic.